R/control_chart.R

```diff
@@ -0,0 +1,139 @@
+#' Create a control chart
+#' 
+#' Create a control chart, aka Shewhart chart:
+#' \url{https://en.wikipedia.org/wiki/Control_chart}. 
+#'
+#' @param d data frame or a path to a csv file that will be read in
+#' @param measure variable of interest mapped to y-axis
+#' @param x variable to go on the x-axis, often a time variable. If unspecified
+#' row indices will be used
+#' @param group1 Optional grouping variable to be panelled horizontally
+#' @param group2 Optional grouping variable to be panelled vertically
+#' @param center_line Function used to calculate central tendency. 
+#' Defaults to mean
+#' @param sigmas Number of standard deviations above and below the central 
+#' tendency to call a point influenced by "special cause variation."
+#' Defaults to 3
+#' @param save_to Optional file path to save chart. If not provided, the chart
+#' will be printed on screen
+#' @param plot_width In inches. Only used if save_to is specified
+#' @param plot_height In inches. Only used if save_to is specified
+#' @param plot_title Title in upper-left
+#' @param plot_catpion Caption in lower-right
+#' @param plot_font_size Base font size; text elements will be scaled to this
+#'
+#' @return Generally called for the side effect of printing the control chart
+#' or writing the control chart to file, depending on whether save_to is 
+#' specified. Invisibly, returns a ggplot object for further customization.
+#' @export
+#' @import ggplot2
+#' @importFrom readr read_csv
+#' @importFrom tibble data_frame
+#' @importFrom stats as.formula
+#'
+#' @examples
+#' d <- 
+#'   tibble::data_frame(
+#'     day = sample(c("Monday", "Tuesday", "Wednesday", "Thursday", "Friday"), 
+#'                  100, TRUE),
+#'     person = sample(c("Tom", "Jane", "Alex"), 100, TRUE),
+#'     count = rbinom(100, 20, ifelse(day == "Friday", .5, .2)),
+#'     date = Sys.Date() - sample.int(100))
+#' control_chart(d, "count")
+#' control_chart(d, "count", group1 = "day", group2 = "person")
+#' if (require(ggplot2)) {
+#' x <- control_chart(d, "count", "date")
+#' x + 
+#'   ylab("Number of Adverse Events") + 
+#'   scale_x_date(name = "Week of", date_breaks = "week") +
+#'   theme(axis.text.x = 
+#'                    element_text(angle = -90, vjust = 0.5, hjust=1))
+#' }
+control_chart <- function(d, measure, x, group1, group2,
+                          center_line = mean, sigmas = 3,
+                          save_to, plot_width = 8, plot_height = 4,
+                          plot_title = NULL, plot_catpion = NULL,
+                          plot_font_size = 11) {
+
+  theme_set(theme_bw())
+
+  if (missing(d) || !(is.data.frame(d) || is.character(d))) {
```

Comment:
This block needs to be tested with the three conditions in the if statement.

Reply:
I think you're missing a `!` in front of `is.character(d)`. This will stop if someone passes a string location of the data file. Whoops, missed the paren. Move along, nothing to see here.

Reply:
And this error handling should probably be put into a function, as I imagine that many other functions are going to take the same style data or path input.

Reply:
Need a check and warning if groups are included by don't exist.

Reply:
@mmastand I don't think we want to let users provide csv filepaths in general. I added it here in case there's someone that doesn't use R at all and just wants this plot. In general putting this kind of error checking in functions is a great idea.